controller/konnect/watch_kongroute.go

```diff
@@ -0,0 +1,137 @@
+package konnect
+
+import (
+	"context"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	operatorerrors "github.com/kong/gateway-operator/internal/errors"
+	"github.com/kong/gateway-operator/modules/manager/logging"
+
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+)
+
+// TODO(pmalek): this can be extracted and used in reconciler.go
+// as every Konnect entity will have a reference to the KonnectAPIAuthConfiguration.
+// This would require:
+// - mapping function from non List types to List types
+// - a function on each Konnect entity type to get the API Auth configuration
+//   reference from the object
+// - lists have their items stored in Items field, not returned via a method
+
+// KongRouteReconciliationWatchOptions returns the watch options for
+// the KongRoute.
+func KongRouteReconciliationWatchOptions(
+	cl client.Client,
+) []func(*ctrl.Builder) *ctrl.Builder {
+	return []func(*ctrl.Builder) *ctrl.Builder{
+		func(b *ctrl.Builder) *ctrl.Builder {
+			return b.For(&configurationv1alpha1.KongRoute{},
+				builder.WithPredicates(
+					predicate.NewPredicateFuncs(kongRouteRefersToKonnectControlPlane(cl)),
+				),
+			)
+		},
+		func(b *ctrl.Builder) *ctrl.Builder {
+			return b.Watches(
+				&configurationv1alpha1.KongService{},
+				handler.EnqueueRequestsFromMapFunc(
+					enqueueKongRouteForKongService(cl),
+				),
```

Comment:
Should we also add predicates to filter `KongService`s that refers to a `KonnectControlPlane` as what we does for `KongRoute`s?

Reply:
Added this check in d4c81205a1e4d300b251b8e503e0060946070afe.

PTAL.